src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12468,7 +12469,18 @@ void Unit::ApplyAttackTimePercentMod(WeaponAttackType att, float val, bool apply
         if (GetTypeId() == TYPEID_PLAYER)
         {
             if (att == BASE_ATTACK)
+            {
                 ApplyPercentModFloatValue(UNIT_FIELD_MOD_HASTE, val, !apply);
+
+                const Unit::AuraEffectList list = GetAuraEffectsByType(SPELL_AURA_MOD_SPELL_COOLDOWN_BY_HASTE);
+
+                for (Unit::AuraEffectList::const_iterator itr = list.begin(); itr != list.end(); ++itr)
```

Comment:
We don't use auto for non-iterator types, also AuraEffectList is a std::list<AuraEffect*>, there is no need to have a reference to pointer there